src/services/pool/concerns/apr.concern.ts

```diff
@@ -0,0 +1,161 @@
+import { isStablePhantom } from '@/composables/usePool';
+import { FiatCurrency } from '@/constants/currency';
+import { bnSum, bnum } from '@/lib/utils';
+import { calcUSDPlusWeightedAPR } from '@/lib/utils/apr.helper';
+import { includesWstEth } from '@/lib/utils/balancer/lido';
+import { aaveService } from '@/services/aave/aave.service';
+import {
+  AnyPool,
+  AprRange,
+  PoolAPRs
+} from '@/services/balancer/subgraph/types';
+import { TokenPrices } from '@/services/coingecko/api/price.service';
+import { lidoService } from '@/services/lido/lido.service';
+import { GaugeBalApr } from '@/services/staking/staking-rewards.service';
+
+export class AprConcern {
+  constructor(
+    public readonly pool: AnyPool,
+    public readonly lido = lidoService,
+    public readonly aave = aaveService
+  ) {}
+
+  public async calc(
+    poolSnapshot: AnyPool | undefined,
+    prices: TokenPrices,
+    currency: FiatCurrency,
+    protocolFeePercentage: number,
+    stakingBalApr: GaugeBalApr,
+    stakingRewardApr = '0'
+  ): Promise<PoolAPRs> {
+    const swapFeeAPR = this.calcSwapFeeAPR(poolSnapshot, protocolFeePercentage);
+    const yieldAPR = await this.calcYieldAPR(
+      prices,
+      currency,
+      protocolFeePercentage
+    );
+    const baseTotalAPR = bnSum([
+      swapFeeAPR,
+      yieldAPR.total,
+      stakingRewardApr
+    ]).toString();
```

Comment:
Note to self, baseTotalAPR probably shouldn't include staking rewards since we will need to reference totals with and without staking. Adjust the schema to make these two scenarios easier to handle.

Reply:
This has been handled